backend/store/principal.go

```diff
@@ -92,14 +99,39 @@ type UserMessage struct {
 	Type                       api.PrincipalType
 	PasswordHash               string
 	IdentityProviderResourceID *string
-	// IdentityProviderUserInfo is a raw json string from userinfo api response.
+	// IdentityProviderUserInfo contains specific fields and a raw json string from userinfo api response.
 	// e.g. for GitHub's authenticated user api, it looks like the following structure.
```

Comment:
This is a bit challenging for callers to use especially without an example. Recommend to use a query map  
map[string]string such as {"user_id": "hello"} for finding the user_id with value hello.

Reply:
use `identifier` instead of SQL.